hack/cli-testing-environment.sh

```diff
@@ -113,6 +113,18 @@ ${BUILD_PATH}/karmadactl init --kubeconfig=${KUBECONFIG_PATH}/${HOST_CLUSTER_NAM
 
 # join cluster
 echo "Join member clusters..."
-${BUILD_PATH}/karmadactl --kubeconfig ${HOME}/karmada/karmada-apiserver.config  join ${MEMBER_CLUSTER_1_NAME} --cluster-kubeconfig=${KUBECONFIG_PATH}/${MEMBER_CLUSTER_1_NAME}.config
-${BUILD_PATH}/karmadactl --kubeconfig ${HOME}/karmada/karmada-apiserver.config  join ${MEMBER_CLUSTER_2_NAME} --cluster-kubeconfig=${KUBECONFIG_PATH}/${MEMBER_CLUSTER_2_NAME}.config
-kubectl wait --for=condition=Ready clusters --all --timeout=800s  --kubeconfig=${HOME}/karmada/karmada-apiserver.config
+TOKEN_CMD=$(${BUILD_PATH}/karmadactl --kubeconfig ${HOME}/karmada/karmada-apiserver.config token create --print-register-command)
+TOKEN=$(echo "$TOKEN_CMD" | grep -o '\--token [^ ]*' | cut -d' ' -f2)
+HASH=$(echo "$TOKEN_CMD" | grep -o '\--discovery-token-ca-cert-hash [^ ]*' | cut -d' ' -f2)
+ENDPOINT=$(kubectl --kubeconfig ${HOME}/karmada/karmada-apiserver.config config view --minify -o jsonpath='{.clusters[0].cluster.server}' | sed 's|^https://||')
+
+${BUILD_PATH}/karmadactl register ${ENDPOINT} \
+    --token ${TOKEN} \
+    --discovery-token-ca-cert-hash ${HASH} \
+    --discovery-timeout=10m \
+    --kubeconfig=${KUBECONFIG_PATH}/${MEMBER_CLUSTER_1_NAME}.config \
+    --cluster-name=${MEMBER_CLUSTER_1_NAME}
```

Comment:
```suggestion
${BUILD_PATH}/karmadactl register ${ENDPOINT} \
    --token ${TOKEN} \
    --discovery-token-ca-cert-hash ${HASH} \
    --kubeconfig=${KUBECONFIG_PATH}/${MEMBER_CLUSTER_1_NAME}.config \
    --cluster-name=${MEMBER_CLUSTER_1_NAME} \
    --karmada-agent-image "${REGISTRY}/karmada-agent:${VERSION}" \
    --v=4
```
- Remove `--discovery-timeout=10m` because this flag was not used correctly in the `register` command, causing a timeout during the process of waiting for the `karmada - agent` to be ready, which ultimately led to the failure of the CI. I will fix this issue in a separate PR.
- add `--v=4` to show more logs
- add `--karmada-agent-image "${REGISTRY}/karmada-agent:${VERSION}"` to specify the karmada-agent-image
Therefore, an additional line of code needs to be added when using `kind` to load the image. like:
https://github.com/karmada-io/karmada/blob/80d3d9e007b59e7d1db4cfb139f73d48fc7458d2/hack/cli-testing-environment.sh#L97-L101
```go
kind load docker-image "${REGISTRY}/karmada-agent:${VERSION}" --name="${MEMBER_CLUSTER_1_NAME}"
```

Reply:
thanks for pointing this out! super interesting!